python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -367,30 +368,125 @@ class RandomForestClassifier(Base):
+        treelite_handle = self._obtain_treelite_handle()
+        treelite_model = \
+            TreeliteModel.from_treelite_model_handle(treelite_handle)
+        return treelite_model
+
+    def convert_to_fil_model(self, output_class=True,
+                             threshold=0.5, algo='auto',
+                             fil_sparse_format='auto'):
+        """
+        Create a Forest Inference (FIL) model from the trained cuML
+        Random Forest model.
+
+        Parameters
+        ----------
+        output_class: boolean (default = True)
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            If true, return a 1 or 0 depending on whether the raw
+            prediction exceeds the threshold. If False, just return
+            the raw prediction.
+        algo : string (default = 'auto')
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            'naive' - simple inference using shared memory
+            'tree_reorg' - similar to naive but trees rearranged to be more
+                           coalescing-friendly
+            'batch_tree_reorg' - similar to tree_reorg but predicting
+                                 multiple rows per thread block
+            `auto` - choose the algorithm automatically. Currently
+                     'batch_tree_reorg' is used for dense storage
+                     and 'naive' for sparse storage
+        threshold : float (default = 0.5)
+            Threshold used for classification. Optional and required only
+            while performing the predict operation on the GPU.
+            It is applied if output_class == True, else it is ignored
+        fil_sparse_format : boolean or string (default = auto)
+            This variable is used to choose the type of forest that will be
+            created in the Forest Inference Library. It is not required
+            while using predict_model='CPU'.
+            'auto' - choose the storage type automatically
+                     (currently True is chosen by auto)
+             False - create a dense forest
+             True - create a sparse forest, requires algo='naive'
+                    or algo='auto'
+        Returns
+        ----------
+        fil_model :
+           A Forest Inference model which can be used to perform
+           inferencing on the random forest model.
+        """
+
+        if fil_sparse_format == 'auto':
+            storage_type = fil_sparse_format
+        elif not fil_sparse_format:
+            storage_type = 'DENSE'
+        elif fil_sparse_format:
+            storage_type = 'SPARSE'
+        else:
+            raise ValueError("The value entered for spares_forest is not "
+                             "supported. Please refer to the documentation "
+                             "to see the accepted values.")
+
+        if (self.max_depth > 16 and (storage_type == 'DENSE' or
```

Comment:
This text is pretty long and the condition is complex with embedded constants... I think we should break this out to a shared

`_check_fil_parameter_validity(depth, storage_format, algo)` function so we dont have to repeat it.